common/src/main/java/com/microsoft/identity/common/internal/ui/embeddedwebview/OAuth2WebViewClient.java

```diff
@@ -57,18 +63,121 @@ public Context getContext() {
+    }
+
+    @Override
+    public void onReceivedHttpAuthRequest(WebView view, final HttpAuthHandler handler,
+                                          String host, String realm) {
+        // Create a dialog to ask for credentials and post it to the handler.
+        Logger.info(TAG, "Receive the http auth request. Start the dialog to ask for creds. ");
+        Logger.infoPII(TAG, "Host:" + host);
+
+        //TODO TelemetryEvent.setNTLM(true); after the Telemetry is finished in common.
+        final NtlmChallengeHandler ntlmChallengeHandler
+                = new NtlmChallengeHandler(view, handler, host, realm, mContext, mCompletionCallback);
+        ntlmChallengeHandler.process();
+    }
+
+    @Override
+    public void onReceivedError(final WebView view,
+                                final int errorCode,
+                                final String description,
+                                final String failingUrl) {
+        super.onReceivedError(view, errorCode, description, failingUrl);
+
+        // Create result intent when webView received an error.
+        final Intent resultIntent = new Intent();
+        resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_ERROR_CODE,
+                "Error Code:" + errorCode);
+        resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE,
+                description);
+        resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_REQUEST_INFO,
+                mRequest);
+
+        // Send the result back to the calling activity
+        mCompletionCallback.sendResponse(AuthenticationConstants.UIResponse.BROWSER_CODE_ERROR, resultIntent);
+    }
+
+    @Override
+    public void onReceivedSslError(final WebView view,
+                                   final SslErrorHandler handler,
+                                   final SslError error) {
+        // Developer does not have option to control this for now
+        super.onReceivedSslError(view, handler, error);
+        handler.cancel();
+
+        // WebView received the ssl error and create the result intent.
+        final Intent resultIntent = new Intent();
+        resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_ERROR_CODE,
+                "Code:" + ERROR_FAILED_SSL_HANDSHAKE);
+        resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE,
+                error.toString());
+        resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_REQUEST_INFO,
+                mRequest);
+
+        // Send the result back to the calling activity
+        mCompletionCallback.sendResponse(AuthenticationConstants.UIResponse.BROWSER_CODE_ERROR, resultIntent);
+    }
+
+    @Override
+    public void onPageFinished(final WebView view,
+                               final String url) {
+        super.onPageFinished(view, url);
+        // Once web view is fully loaded,set to visible
+        view.setVisibility(View.VISIBLE);

```

Comment:
Is this view not visible by default?

Reply:
No, we set the webview `INVISIBLE` when creating it, and set it `VISIBLE` once the webview finish loading.